src/components/clusters/ClustersFilterToolbar.tsx

```diff
@@ -0,0 +1,141 @@
+  Button,
+  ButtonVariant,
+  InputGroup,
+  Select,
+  SelectOption,
+  TextInput,
+  ToolbarProps,
+  ToolbarFilterProps,
+  SelectProps,
+  TextInputProps,
+} from '@patternfly/react-core';
+import { SearchIcon, FilterIcon } from '@patternfly/react-icons';
+import { Cluster } from '../../api/types';
+import { CLUSTER_STATUS_LABELS } from '../../config';
+
+export type ClusterFiltersType = {
+  [key: string]: string[]; // value from CLUSTER_STATUS_LABELS
+};
+
+type ClustersFilterToolbarProps = {
+  searchName: string;
+  setSearchName: (value: string) => void;
+  filters: ClusterFiltersType;
+  setFilters: (filters: ClusterFiltersType) => void;
+};
+
+const ClustersFilterToolbar: React.FC<ClustersFilterToolbarProps> = ({
+  searchName,
+  setSearchName,
+  filters,
+  setFilters,
+}) => {
+  const [isStatusExpanded, setStatusExpanded] = React.useState(false);
+
+  const onClearAllFilters: ToolbarProps['clearAllFilters'] = () => {
+    setFilters({
+      status: [],
+    });
+  };
+
+  const onSearchNameChanged: TextInputProps['onChange'] = setSearchName;
+
+  const onSelect = (type: string, isChecked: boolean, value: Cluster['status']) => {
+    setFilters({
+      ...filters,
+      [type]: isChecked
+        ? [...filters[type], value]
+        : filters[type].filter((v: string) => v !== value),
+    });
+  };
+
+  const onStatusToggle: SelectProps['onToggle'] = () => setStatusExpanded(!isStatusExpanded);
+  const onStatusSelect: SelectProps['onSelect'] = (event, value) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
+    onSelect('status', event.target.checked, value as string);
+  };
+
+  const onDeleteChip: ToolbarFilterProps['deleteChip'] = (type, id) => {
+    if (type) {
+      setFilters({
+        ...filters,
+        [type as string]: filters[type as string].filter((v: string) => v !== id),
+      });
+    } else {
+      onClearAllFilters();
+    }
+  };
+
+  const onDeleteChipGroup: ToolbarFilterProps['deleteChipGroup'] = (type) => {
+    setFilters({
+      ...filters,
+      [type as string]: [],
+    });
+  };
+
+  const statusPlaceholder = (
+    <>
+      <FilterIcon /> Status
+    </>
+  );
+
+  return (
+    <Toolbar
+      id="clusters-filter-toolbar"
+      className="pf-m-toggle-group-container"
+      collapseListedFiltersBreakpoint="xl"
+      clearAllFilters={onClearAllFilters}
+    >
+      <ToolbarContent>
+        <ToolbarFilter
+          chips={filters.status}
+          deleteChip={onDeleteChip}
+          deleteChipGroup={onDeleteChipGroup}
+          categoryName="status"
+        >
+          <Select
+            variant="checkbox"
+            aria-label="status"
+            onToggle={onStatusToggle}
+            onSelect={onStatusSelect}
+            selections={filters.status}
+            isOpen={isStatusExpanded}
+            placeholderText={statusPlaceholder}
+          >
+            {Object.getOwnPropertyNames(CLUSTER_STATUS_LABELS).map((status) => (
```

Comment:
```suggestion
            {Object.keys(CLUSTER_STATUS_LABELS).map((status) => (
```
seems to work as well

Reply:
there is no difference in our case, but changed